src/sbuf/MemBlob.h

```diff
@@ -91,9 +91,18 @@ class MemBlob: public RefCountable
      */
     void append(const char *source, const size_type n);
 
+    /* non-const methods below require exclusive object ownership */
+
     /// extends the available space to the entire allocated blob
     void clear() { size = 0; }
 
+    /// keep the first n bytes and forget the rest of data
+    void syncSize(const size_type n) { size = n; }
```

Comment:
Added in f698a23. I did not add it earlier because clear() does not have it, `size` is public, and the method is inlined. A proper solution would be a smart pointer interface that does not allow write access without locking (and pays the checking price once), but that is out of scope here.

Reply:
This addition [broke](https://semaphoreci.com/squid/squid/branches/pull-request-744/builds/4) the build. When I get a chance, I will fix it by de-inlining the syncSize() method.

Reply:
Done in 67b4a95.